networkx/algorithms/tree/mst.py

```diff
@@ -721,6 +722,228 @@ def maximum_spanning_tree(G, weight="weight", algorithm="kruskal", ignore_nan=Fa
     return T
 
 
+@py_random_state(3)
+def random_spanning_tree(G, weight=None, multiplicative=True, seed=None):
```

Comment:
Since this is new API, what do we think about making the optional args keyword-only? I'd advocate for:

```suggestion
def random_spanning_tree(G, *, weight=None, multiplicative=True, seed=None):
```

but there's an argument for keeping `weight` varpos if others prefer that.

Reply:
I personally think that leaving `weight` as a positional argument makes more sense and I think will be more complaint with the existing networkx interface, but the other two (`multiplicative` and `seed`) can totally be made keyword-only.

Reply:
That seems sensible to me, especially since this would most commonly result in something like `weight="weight"`, which is not really a *huge* improvement in terms of readability. I'm +1 on this. Thoughts @MridulS @dschult ?

Reply:
This makes sense to me too.  `weight` doesn't need to be keyword. But the others should be.